include/dip.hpp

```diff
@@ -0,0 +1,82 @@
+//===- dip.hpp --------------------------------------------------------===//
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===----------------------------------------------------------------------===//
+//
+// Header file for DIP dialect specific operations and other entities.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef INCLUDE_DIP
+#define INCLUDE_DIP
+
+// Define Memref Descriptor.
+typedef struct MemRef_descriptor_ *MemRef_descriptor;
```

Comment:
Now we can use the MemRef descriptor like this for a temporary solution. I think we should split the MemRef as a separate module (like MemRef in buddy-benchmark) in the future. So can we add a TODO here for future work?